osu.Game/Beatmaps/ControlPoints/ControlPoint.cs

```diff
@@ -2,13 +2,15 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
+using System.Diagnostics.CodeAnalysis;
 using Newtonsoft.Json;
 using osu.Game.Graphics;
 using osu.Game.Utils;
 using osuTK.Graphics;
 
 namespace osu.Game.Beatmaps.ControlPoints
 {
+    [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
```

Comment:
Have you tested that this is enough, on the real thing, rather than on the example in the OP? On a quick look `[DynamicallyAccessedMembers]` has [`Inherited = false` specified](https://github.com/dotnet/runtime/blob/257e76dfe64cb5b347aba7bc5a31a50c3c8cc106/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/DynamicallyAccessedMembersAttribute.cs#L32) so I am not immediately confident that it will fully eliminate your workaround.

Reply:
I admit I initially didn't, because it's kinda annoying to, which is why I did it in a separate example.

It does look to be working in practice though, with one caveat - I missed that `LegacyControlPointInfo` isn't a `ControlPoint` (naming... >_<) Will fix in next commit.

Reply:
Fyi, I think that this part of the documentation is the most relevant: https://github.com/dotnet/runtime/blob/257e76dfe64cb5b347aba7bc5a31a50c3c8cc106/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/DynamicallyAccessedMembersAttribute.cs#L19-L21

I also noticed `Inherited = false` and questioned this behaviour, and had to verify myself. I also explored the alternative of using `System.Diagnostics.CodeAnalysis.DynamicDependencyAttribute` but _that one_ does not work for inherited types.